src/scipp/core/cumulative.py

```diff
@@ -5,12 +5,14 @@
 from typing import Literal, Optional
 
 from .._scipp import core as _cpp
+from ..typing import VariableLikeType
 from ._cpp_wrapper_util import call_func as _call_cpp_func
+from .concepts import transform_data
 
 
-def cumsum(a: _cpp.Variable,
+def cumsum(a: VariableLikeType,
            dim: Optional[str] = None,
-           mode: Literal['exclusive', 'inclusive'] = 'inclusive') -> _cpp.Variable:
+           mode: Literal['exclusive', 'inclusive'] = 'inclusive') -> VariableLikeType:
```

Comment:
Annotations with `VariableLikeType` produce hard to read output in sphinx. I used an explicit type in the docstring, see e.g. `broadcast`. But that makes the docs less precise. So I leave it up to you to pick one.